### PR TITLE
[CompilerPlugin] Factor out message handling logic to a module

### PR DESCRIPTION
**Pull request by rintaro**
Create a module `_SwiftCompilerPluginImpl` that abstracts the low level message connection/serialization.

`CompilerPlugin` is now an implementation of the message connection, and adapts the concrete type to the message handler.

This make it possible to make a plugin with a different messaging connection and serialization.

### CODE REVIEW

**Comment by rintaro** [comment]
@swift-ci Please smoke test

**Comment by rintaro** [comment]
@swift-ci Please test

**Review by bnbarham — APPROVED** [review]
Not a huge fan of `Impl`. `CompilerPluginAPI` or something?

File: Package.swift [hunk 1]
```diff
@@ -43,6 +43,7 @@ let package = Package(
     .library(name: "SwiftSyntaxParser", type: .static, targets: ["SwiftSyntaxParser"]),
     .library(name: "SwiftSyntaxBuilder", type: .static, targets: ["SwiftSyntaxBuilder"]),
     .library(name: "SwiftSyntaxMacros", type: .static, targets: ["SwiftSyntaxMacros"]),
+    .library(name: "_SwiftCompilerPluginBase", type: .static, targets: ["_SwiftCompilerPluginImpl"]),
```
Comment (ahoppen) [inline Package.swift:46]: The library is named differently than the target. Is that intentional?

**Review by ahoppen — COMMENTED** [review]
What do you think about naming the module `SwiftCompilerPluginMessageHandling` or something like that. I think that’s more descriptive than `Impl`. Also, I’m not sure whether we should underscore it because it seems equally useful to users as the other modules in swift-syntax.

**Comment by rintaro** [comment]
Thank for naming idea. Let me go with `SwiftCompilerPluginMessageHandling`

**Pull request merged**